### DISCUSSION TITLE
Local backup to test

### DISCUSSION THREAD

**Discussion by GladysSobrido** [apostrophecms/apostrophe#3213] [Q&A] [answered]
Hello, I'm new to Apostrophe but as I have used other CMS in the past (joomla), my friend has given me the credentials to their ApostropheCMS so I can try to fix some issues on their website. I would like to make a local copy of the full site on my computer so I can do tests over there before making any permanent changes on the real website. Can you tell me how can I do this backup? thank you.

**Comment by abea** [comment, marked as answer]
I assume you're talking about copying the data and you already have access to the site code. If you're asking about setting up the site code, you should see the [environment setup](https://v2.docs.apostrophecms.org/getting-started/setting-up-your-environment.html) documentation.

Copying the content could be easy or... less easy. If they used the official boilerplate then it should have a [sync down script](https://github.com/apostrophecms/apostrophe-boilerplate/blob/main/scripts/sync-down) already. It would also depend on them having populated a `/deployment/settings.production` file with production server IP and username. To use that script you would run `bash ./scripts/sync-down production` from the command line locally when at the project root. It would probably ask for the server user password unless you have a matching SSH key.

If that isn't set up already you could set it up. Again, that involves setting up the production deployment settings file following the boilerplate [example](https://github.com/apostrophecms/apostrophe-boilerplate/blob/main/deployment/settings.production.example) and making sure that the sync-down script is in the project.

Alternatively you could manually pull the uploaded files and MongoDB database from the server to your local machine. The best way to see those steps really is to [look at the sync down script](https://github.com/apostrophecms/apostrophe-boilerplate/blob/main/scripts/sync-down). The important lines start on line 28.

If that isn't the information you needed, please update with more specific details.

**Reply by GladysSobrido** [reply]
Thank you for your answer.  I don't have access to the code by now. If I install the enviroment on my computer, will I be able to acess the code then? Do I need the credentials of the hosting provider?
Thank you

**Reply by abea** [reply]
The code for their website? You would need to ask them where that is. I would hope in version control somewhere, but you could also use rsync or FTP to pull the code from their server. It sounds like you need to learn more about their web hosting and the project generally before proceeding.

**Reply by GladysSobrido** [reply]
The website was created by a man with who they didn't had a good relationship in the end, so he didn't left any document explaining how he had created it. It was supposed to be a simple clean site but he mixed Apostrophe with something else, I think he inserted code modules, now there are things as pictures or a foot with a menu that cannot be modified though Apostrophe.
 I would try to download it all with FTP and find out a way to modify it.
Thank you

**Comment by boutell** [comment]
I see. You'll want to read the Apostrophe Getting Started guide then, and
you'll need to learn how to use the mongodump, mongorestore and rsync
utilities to get a copy of the database on your local computer.

If you have some budget for this, you might want to consider reaching out
to an ApostropheCMS partner company to see if they are up for helping
you with this project:

https://apostrophecms.com/partners

On Thu, Jul 1, 2021 at 3:11 PM GladysSobrido ***@***.***>
wrote:

> The website was created by a man with who they didn't had a good
> relationship in the end, so he didn't left any document explaining how he
> had created it. It was supposed to be a simple clean site but he mixed
> Apostrophe with something else, I think he inserted code modules, now there
> are things as pictures or a foot with a menu that cannot be modified though
> Apostrophe.
> I would try to download it all with FTP and find out a way to modify it.
> Thank you
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/apostrophe/discussions/3213#discussioncomment-952373>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27O5GQ3PIYDDZ6BVHLTTVS4UVANCNFSM47U7BOBQ>
> .
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his
